2022-10-20 Code Review

wkbraid opened the pull request.

Nolan-547 committed:
Initial commit

Generated by create-expo-app 1.1.1.

Nolan-547 committed:
Initial commit

Generated by create-expo-app 1.1.1.

Nolan-547 committed:
This is the first commit

Nolan-547 committed:
This is the first commit

nolan-3 committed:
second message

nolan-3 committed:
second message

nolan-3 committed:
second commit

nolan-3 committed:
second commit

nolan-3 committed:
second message

nolan-3 committed:
second message

nolan-3 committed:
Merge branch 'main' of https://github.com/nolan-3/Dodgeball

nolan-3 committed:
Merge branch 'main' of https://github.com/nolan-3/Dodgeball

nolan-3 committed:
code not using classes that doesn't work

nolan-3 committed:
code not using classes that doesn't work

nolan-3 committed:
Several bouncing projectiles

nolan-3 committed:
Several bouncing projectiles

nolan-3 committed:
sized window correctly and projectiles nor bounce off the player

nolan-3 committed:
sized window correctly and projectiles nor bounce off the player

nolan-3 committed:
used command: 'npm expo prebuild' to try to make the native code accessable for xcode so I can run on my phone

nolan-3 committed:
used command: 'npm expo prebuild' to try to make the native code accessable for xcode so I can run on my phone

nolan-3 committed:
1. Changed Knight to a functional component so that I could call an event listener for key input, however now it doesn't run in the gameloop so it's quite clunky. 2. Downloaded Xcode and tried to convert this into native code for android/ios, I don't think it worked. Finally I would just like to say I think this solution for moving is quite poor, maybe making it back into an entity and having it rerender each time the loop runs would work. I just kept getting an error Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:

You might have mismatching versions of React and the renderer (such as React DOM)
You might be breaking the Rules of Hooks
You might have more than one copy of React in the same app See for tips about how to debug and fix this problem. I believe it was number 2 and I wasn't sure how to work around it.

nolan-3 committed:
1. Changed Knight to a functional component so that I could call an event listener for key input, however now it doesn't run in the gameloop so it's quite clunky. 2. Downloaded Xcode and tried to convert this into native code for android/ios, I don't think it worked. Finally I would just like to say I think this solution for moving is quite poor, maybe making it back into an entity and having it rerender each time the loop runs would work. I just kept getting an error Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:

You might have mismatching versions of React and the renderer (such as React DOM)
You might be breaking the Rules of Hooks
You might have more than one copy of React in the same app See for tips about how to debug and fix this problem. I believe it was number 2 and I wasn't sure how to work around it.

nolan-3 committed:
1. Changed Knight to a functional component so that
I could call an event listener for key input, however now it doesn't run in the gameloop so it's
quite clunky. 2. Downloaded Xcode and tried to convert this into native code for android/ios, I d
on't think it worked. Finally I would just like to say I think this solution for moving is quite
poor, maybe making it back into an entity and having it rerender each time the loop runs would wo
rk. I just kept getting an error Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:

You might have mismatching versions of React and the renderer (such as React DOM)
You might be breaking the Rules of Hooks
You might have more than one copy of React in the same app See for tips about how to debug and fix this problem. I believe it was number 2 and I wasn't sure how to work around it.

nolan-3 committed:
1. Changed Knight to a functional component so that
I could call an event listener for key input, however now it doesn't run in the gameloop so it's
quite clunky. 2. Downloaded Xcode and tried to convert this into native code for android/ios, I d
on't think it worked. Finally I would just like to say I think this solution for moving is quite
poor, maybe making it back into an entity and having it rerender each time the loop runs would wo
rk. I just kept getting an error Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:

You might have mismatching versions of React and the renderer (such as React DOM)
You might be breaking the Rules of Hooks
You might have more than one copy of React in the same app See for tips about how to debug and fix this problem. I believe it was number 2 and I wasn't sure how to work around it.

nolan-3 committed:
hi

nolan-3 committed:
hi

nolan-3 committed:
added, at least visually, the joystick, seems like it should be easy enough to use for mobile, however getting to the point where I can test it on mobile might be a bit more difficult.

nolan-3 committed:
added, at least visually, the joystick, seems like it should be easy enough to use for mobile, however getting to the point where I can test it on mobile might be a bit more difficult.

wkbraid committed:
Test commit, fixed some indentation.

wkbraid committed:
Test commit, fixed some indentation.

nolan-3 committed:
knight moves smoothly now

nolan-3 committed:
knight moves smoothly now

nolan-3 committed:
knight moves smoothly now

nolan-3 committed:
knight moves smoothly now

nolan-3 committed:
increased knight max speed and decreased time to reach max speed, added a circle projectile which follows the player and the same logic to check for collisions.

nolan-3 committed:
increased knight max speed and decreased time to reach max speed, added a circle projectile which follows the player and the same logic to check for collisions.

nolan-3 committed:
added 'homepage', set a timer for 10 seconds to return to homepage (eventually will display some screen with how the player did or the result of the overall attack then) made the player slightly smaller, maybe some other things but I'm forgetting

nolan-3 committed:
added 'homepage', set a timer for 10 seconds to return to homepage (eventually will display some screen with how the player did or the result of the overall attack then) made the player slightly smaller, maybe some other things but I'm forgetting

nolan-3 committed:
changed the creation of entities from loops to functions, added a new type of projectile, I modeled it off a grenade, it uses a timer to determine when to explode

nolan-3 committed:
changed the creation of entities from loops to functions, added a new type of projectile, I modeled it off a grenade, it uses a timer to determine when to explode

nolan-3 committed:
changed the creation of entities from loops to functions, added a new type of projectile, I modeled it off a grenade, it uses a timer to determine when to explode

nolan-3 committed:
changed the creation of entities from loops to functions, added a new type of projectile, I modeled it off a grenade, it uses a timer to determine when to explode

nolan-3 committed:
Results screen actually works now, Next I am going to figure out how to reset the game, I kind of figured it would be automatic but it's clearly not

nolan-3 committed:
Results screen actually works now, Next I am going to figure out how to reset the game, I kind of figured it would be automatic but it's clearly not

wkbraid committed:
Added initial empty commit.

nolan-3 committed:
Test Commit

nolan-3 committed:
Test Commit

nolan-3 committed:
Revised code except for eliminating the initial functions

nolan-3 committed:
finished doing your suggestions, albeit I still don't think the key input is well written

nolan-3 committed:
added a few images, It seems that the creating an image needs to be static, eg it can't be passed in as a prop, also noticed that inputting multiple directions can be wonky

nolan-3 committed:
just a visual tweak or two

nolan-3 committed:
images

nolan-3 committed:
better bouncing off walls for the bricks, I don't think I want the other projectiles to bounce so I will leave them be, albeit if I do the exact same idea should work

wkbraid commented on Dodge.js:
```diff
@@ -0,0 +1,179 @@
+//adds an entity given its type, if it's a Grenade set a timer for explosion
+function AddEntity(entity){
+let id = generateUniqueId()
+ents[id] = new entity(id)
```
This is interesting, but it would probably be better for this function to accept an already constructed entity. i.e.

In AddEntity function:
```
function AddEntity(entity) {
  // Assume all entities have an id
  ents[entity.id] = entity; // Just store the value
}
```

In the entity class definition:
```
class Grenade {
  constructor() {
     // In my opinion: Nothing wrong for entities not in the list to have ids, maybe even useful for debugging.
     this.id = generateUniqueId();
  }
}
```

Elsewhere, when you add an entity:
```
// Make a new grenade, and keep track of it.
AddEntity(new Grenade(50, 100));
```

wkbraid replied:
This gives you some flexibility too, e.g. different classes could have different styles of unique IDs for readability:
```
class Knight {
  constructor() {
    // As long as there really is only one "Knight", we can make it easy to lookup
    // Elsewhere: ents["knight"] will refer to this object.
    this.id = "knight"
  }
}
```

```
class Grenade {
  constructor() {
     // Still unique, but it's easier to tell what kind of object it is at a glance.
     // It's probably not good to rely on this for anything but debugging.
     // If you want to tell what type of object something is, I suggest having another field for that explicitly.
     this.id = "grenade-" + generateUniqueId();
  }
}
```

wkbraid commented on Dodge.js:
```diff
@@ -0,0 +1,179 @@
+let id = generateUniqueId()
+ents[id] = new entity(id)
+
+if(entity == Grenade){
```
[Bigger Refactor] This logic should really be inside the Grenade class.

I assume the problem is that within the Grenade class, it's not clear how to call Explode since it refers to `ents`.
I think this will happen fairly often, e.g. you run into the same problem with tracking the Knights position.

I think you want each entity to be able to access the 'ents' variable. In fact, maybe you want each entity to be able to access all the functions in this file. How can you do that?

I suggest making a class to contain everything about the DodgeBall game, i.e. most of the contents of the file we're in right now:
```
export default class DodgeballGame extends PureComponent {
  constructor({navigation}) {
    // Define "global" variables (which are now variables for this object)
    this.ents = {}
    this.AddEntity(new Circle())
    this.AddEntity(new Knight())
  }

  renderer() {
   // defines how to display a "DogeballGame"
    return (
      <View>
        <GameEngine>
           ...
        </GameEngine>
      </View>
    )
  }

  // What used to be functions are now methods, i.e. you're adding an entity to a particular DodgeBallGame
  DeleteEntity(entity) {
    delete(this.ents[entity.id]);
  }

  AddEntity(entity) {
    // ** HERE **
    // Now that DodgeBallGame is an object, we can pass a reference to the whole thing to each entity
    // allowing it to refer back later.
    entity.parent = this;

    // refer to "this version" of the ents dictionary.
    this.ents[entity.id] = entity;
  }
}
```

Grenade can then handle its own exploding logic:
```
class Grenade {
  constructor() {
    ...
    // All grenades explode 2s after they are created.
    setTimeout(() => { this.Explode() }, 2000);
  }

  Explode() {
     // Tell the game we are a part of to delete our-self, "very meta".
     // NB: this.parent isn't set in the constructor, so you I wouldn't assume it exists.
     //        technically, someone could create a Grenade, and never add it as an entity.
     // NB: NB: There's some nice syntax you could use here called "optional chaining"
     //               this.parent?.InitialShrapnel(this.x, this.y) // note the ?.
     //               will either call InitialShrapnel, or return undefined (doing nothing) if this.parent is undefined
     if (this.parent) {
        this.parent.InitialShrapnel(this.x, this.y);
        this.parent.DeleteEntity(this.id);
     }
  }
}
```

And elsewhere, you could even have two DodgeBallGames at once if you wanted:
```
let dodgeball1 = <DodgeBallGame />
let dodgeball2 = <DodgeBallGame />

// Dodgeball2 is harder, because it has an extra circle.
// I mean, that and because you're playing two games at once.
dodgeball2.AddEntity(new Circle());

return (
  <View>
    {dodgeball1}
    {dodgeball2}
  </View>
);
```

wkbraid commented on Dodge.js:
```diff
@@ -0,0 +1,179 @@
+
+//deletes a single entity given that entities unique id
+function DeleteEntity(id){
+delete(ents[id])
```
[Cosmetic] Indentation, in like a million places :P 
Remember your code editor should be able to "format document" or something like that.

wkbraid commented on Dodge.js:
```diff
@@ -0,0 +1,179 @@
+}
+
+//runs 60 times a second updating each entitiy
+function Gameloop(entities, { touches }, )  {
```
[Cosmetic] The [Google Style Guide](https://google.github.io/styleguide/jsguide.html#naming-class-names) suggests using UpperCamelCase for classes, but lowerCamelCase for functions and methods, to help distinguish between the two. Not necessary, but I do think it's a helpful norm.

nolan-3 replied:
I believe the renderers have to be PascalCase

wkbraid commented on Dodge.js:
```diff
@@ -0,0 +1,179 @@
+}
+
+//creates a square component
+function CreateBrick(props) {
```
You could move the CreateX functions to the Entities file, since that's where they're now used.

We talked about this, but they are all quite similar. You can in fact get them to work with the image passed in:

```
// The classic name for this would probably be "Sprite", basically a generic movable image
function CreateImage(props) {
  return (
    <Image source= {props.image}
       style={{
         width: props.width,
         ...
       }} />
  )
}
```

Apparently, the trick is whatever you require() should be a static source string, so we call require in the constructor and pass the loaded image to CreateImage (not just the url):
```
class Grenade {
  constructor() {
     this.renderer = <CreateImage image={require("./assets/Brick.jpeg")} />
  }
}
```

Why do you need to do it this way? The other code is interpreted as wanting it to dynamically update the image source. i.e.:
```
let x = "./assets/Brick.jpeg";
<Image source={require(x)} />
```
Worries that the value of "x" might change, and it would have to try to dynamically load a different url.

wkbraid commented on Dodge.js:
```diff
@@ -0,0 +1,179 @@
+}
+
+  //creates one shrapnel in each of the cardinal directions
+  function InitialShrapnel(x, y){
```
Maybe this should be moved into the Grenade class? (you'd have to make some changes above first)

Partly just how you think about it, i.e. is it the world, or the grenade that is "responsible" for the shrapnel.
But probably, if there's only one class that's ever likely to use a function, that function should live inside that class.

wkbraid commented on Dodge.js:
```diff
@@ -0,0 +1,179 @@
+}
+
+export default function Dodge({ navigation }) {
+  //this code only runs when navigated to from the home page, why?
```
This is a property of the NavigationStack component you use in App.js.
Basically, it creates each screen *once* (maybe at the start? maybe when it's first displayed?) and then just switches which one is shown to the user. Basically like different tabs.

nolan-3 replied:
Hmm Do I need to include this in the DodgeBall class component (the dodging game)

wkbraid replied:
The direction I was thinking is more like, implement your own screen change:
Have a function in App.js called "goToHomeScreen()" which deletes the current dodgeball game, and changes the display to a home message.
Have another function called "startDodgeBall()" which creates a *new* dodgeball game, then changes the display to show the game.
Have another function called "goToScoreScreen()" which takes a score to display, deletes the current dodgeball game, and changes the display to a screen showing the score.

Rather than passing "navigation" to DodgeBallGame you can pass these functions directly, or maybe an "App" object that has these functions as methods. [Similar to how each entity is passed a "DodgeBallGame", so that it can call functions like "addEntity()" and "deleteEntity()"]

wkbraid commented on Entities.js:
```diff
@@ -0,0 +1,339 @@
+
+//count intersections with players
+const Counter = () => {
+  console.log({counter})
```
Probably move this to Dodge.js, change the logic to update it to be through this.parent.counter.

wkbraid commented on Entities.js:
```diff
@@ -0,0 +1,339 @@
+const MAXwidth = Dimensions.get('window').width;
+const MAXheight = Dimensions.get('window').height;
+let counter = 0
+let GlobalX = 400
```
You should be able to get rid of GlobalX and GlobalY and refer to this.parent.ents["knight"].x

wkbraid commented on Entities.js:
```diff
@@ -0,0 +1,339 @@
+} 
+
+//function to detect wasd input on computer
+function MoveKnight (){
```
Maybe move this to Dodge.js? Have it set ents["knight"].input[0] = 1.5

It does affect an Entity, but its really more about what affect the keyboard input has. e.g. I can imagine re-using the Knight in a different game, where the controls are different (or run by an AI or whatever).

wkbraid replied:
Also, I think the same effect might be more intuitive if you thought of it as "acceleration". i.e.
When "left" is pressed: acceleration[0] -= 1.5; When "left" is released: acceleration[0] += 1.5
When "right" is pressed: acceleration[0] += 1.5; etc
(Your version is technically safer, since it will fix itself if the browser "forgets" a keyup)


[This is kind of overkill, but might be interesting]
Alternatively: have this function just record what interesting keys are currently down:
```
// Listens for interesting key presses, recording which ones are currently down in "input"
// NB: you could even simplify this and just keep a dictionary of *every* key which is currently down
function ListenForKeys(keys) {
  // One entry for each interesting key, they all start "up"
  for (key in keys) {
     input[key] = False
  }
  document.addEventListener('keydown', (event) => {
        if(event.key in input) {  // this is an interesting key
          input[event.key] = True;
        }
   });
   document.addEventListener('keyup', (event) => {
        if(event.key in input) {  // this is an interesting key
          input[event.key] = False;
        }
   });
```

Elsewhere, at the start of the program:
```
ListenForKeys(['w', 'a', 's', 'd']);
```

Elsewhere, maybe in the game loop:
```
if (input['w']) { knight.acceleration[1] -= 1.5; }
```

nolan-3 replied:
I would but it was a huge hassle to get it to work at all, the error culprit:
yOu maY bE BreaKing THe rULEs of hOoKs
I really don't want to deal with that again unless it is necessary, I mean or unless you know what I'm doing wrong I suppose

wkbraid replied:
Yeah I don't think it's interesting or important to struggle with input, it's not surprising to me that it's a place with pain points.

I would still suggest a small change, where maybe instead of input[<number 1 to 4>] you use input[<key that was pressed, e.g. 'w' or 'left'>] just to make the meaning more clear, without changing the functionality.

Later in the move function it becomes
```
knight.velocity.x = knight.velocity.x + input['right'] - input['left']
```

Rather than
```
knight.velocity.x = knight.velocity.x + input[1] - input[0] // how do you know it's not input[2] - input[0] or input[0] - input[1]?
```

wkbraid commented on Entities.js:
```diff
@@ -0,0 +1,339 @@
+    this.backgroundColor = "blue";
+    this.x = 400;
+    this.y = 400;
+    this.velocityX = 0;
```
Either have velocity or momentum, not both :P

wkbraid commented on Entities.js:
```diff
@@ -0,0 +1,339 @@
+    this.velocityY = 0;
+    this.Image = "./assets/Background.jpeg";
+    this.renderer = <CreateBrick/>;
+    this.momentum = [0,0];
```
You could make this an object with names, i.e.:
```
this.momentum = {x: 0, y: 0};
```
Elsewhere:
```
this.momentum.x += 3;
```

nolan-3 replied:
I think I'm fine with leaving it as an array, to me it is pretty clear that it is x and then y

wkbraid commented on Entities.js:
```diff
@@ -0,0 +1,339 @@
+// if x and y are presse
+//update knights position
+update(){
+  let horizontal = input[1] - input[0]
```
At least name these "accelerationX" and "accelerationY" or something like that.

wkbraid replied:
Also put the divided by 4 here, not when adding it to momentum.

wkbraid commented on Entities.js:
```diff
@@ -0,0 +1,339 @@
+update(){
+  let horizontal = input[1] - input[0]
+  let vertical = input[3] - input[2]
+  if(Math.abs(this.momentum[0] + horizontal/4) > 4){
```
Simplier: Add it first, then re-adjust if necessary:
```
this.momentum[0] += accelerationX;
if (Math.abs(this.momentum[0]) > MAXmomentum) { 
  this.momentum[0] = Math.sign(this.momentum[0]) * MAXmomentum
}
```

wkbraid commented on Entities.js:
```diff
@@ -0,0 +1,339 @@
+    this.momentum[1] += vertical/4
+  }
+  //slow the knights velocity by a small amount (to lessen drifting)
+  this.momentum[0] -= .01 * Math.sign(this.momentum[0])
```
Weirdly enough, with this logic, if you were moving veeeery slowly, you would change direction and gain some speed. Probably not actually a problem, but you could add another check to dead stop at zero here.

nolan-3 replied:
haha hopefully that won't be a problem, and as of now I believe it is not possible actually because .01 is the smallest increment/decrement to momentum (and the other is a larger factor of .01)

wkbraid commented on Entities.js:
```diff
@@ -0,0 +1,339 @@
+  this.x += this.momentum[0]
+  this.y += this.momentum[1]
+  //update the knights position and record the change
+  GlobalX = this.x
```
These should go away, see above.

wkbraid commented on Entities.js:
```diff
@@ -0,0 +1,339 @@
+class Brick {
+  constructor(id) {
+    this.id = id;
+    let x = Math.random()*500;
```
Just set this.x = Math.random()*500 directly.

wkbraid commented on Entities.js:
```diff
@@ -0,0 +1,339 @@
+  }
+}
+
+class Circle {
```
There is a lot of overlap and common definitions between Brick and Circle and Grenade (and maybe even Knight).
When you're using Classes, and they have a lot of similar behaviors, one thing you can use is inheritance:
```
// Define some common defaults for all shapes
// You can overwrite these definitions later if you need to
class Shape {
  constructor(x, y) {
    this.x = x;
    this.y = y;
    this.height = 75;
    this.width = 75;
    this.color = "red"
    this.momentum = [0, 0]
  }
  update() {
    // all shapes move and have momentum
    this.x += this.momentum[0];
    this.y += this.momentum[0];
  }
  sayHello() {
    console.log("Hello!")
  }
}

// Now we define a more specific shape, but inherit the definitions above
class Circle extends Shape {
  constructor(x, y) {
    // do common setup:
    super(x, y)
    // do specific updates
    this.renderer = <CreateCirlce />
    // you can just overwrite variables that are defined in the super's constructor:
    this.color = "blue"
  }
  update() {
     // do common update tasks:
     super.update()

     // do some circle specific updates:
     this.color = "green"
  }
}

let circ = new Circle(10, 10);
circ.x // value: 10 (set in super constructor)
circ.update() // first moves, then changes color
circ.sayHello() // prints "Hello!", (uses the "default" definition from the Shape class, even though we didn't define it for Circle)
```

wkbraid commented on Entities.js:
```diff
@@ -0,0 +1,339 @@
+//check if there is a possible for up down
+//check if there is a possible for right left
+//check if those two things overlap
+    if(this.y + this.width < GlobalY  || this.y > GlobalY + playerHeight){
```
"Intersection" is a good candidate for a common function somewhere.
Although as we talked about, you might need to treat circles/rectangles/other shapes differently.

wkbraid commented on Entities.js:
```diff
@@ -0,0 +1,339 @@
+
+}
+reverse() {
+  this.velocityX = (this.velocityX * -1)
```
this.velocityX *= -1; works here.

wkbraid commented on Entities.js:
```diff
@@ -0,0 +1,339 @@
+class Grenade {
+  constructor(id) {
+    this.id = id
+    this.borderRadius = 75;
```
Not actually used?

wkbraid commented on Entities.js:
```diff
@@ -0,0 +1,339 @@
+    this.velocityX += this.velocityX/75
+    this.velocityY += this.velocityY/75
+
+    //preliminary checks to see if the x or y by itself rules out the
```
Cool idea, although it probably doesn't do that much in this case (and technically it could hurt, since now every time there *isn't* a collision you're adding an extra step)

For general advice: Don't worry too much about optimization at first. Use a tool that will help identify the slowest parts of your code (normally: which functions do I spend the most time inside, so not just slow functions, but commonly called ones matter most). Then go and optimize that part. Repeat if necessary.

wkbraid replied:
These are normally called "profilers"

wkbraid commented on Entities.js:
```diff
@@ -0,0 +1,339 @@
+    let x1 = GlobalX
+    let y1 = GlobalY
+
+    //check if its above or below
```
I'm not sure why this logic is here, maybe this is trying to check whether a Circle overlaps with a Rectangle?

nolan-3 replied:
yeah, although the logic isn't there yet (or anywhere else for that matter)

wkbraid commented on README.md:
```diff
@@ -0,0 +1,14 @@
+To run these commands, you may need to install [git](https://github.com/git-guides/install-git), [npm,](https://www.npmjs.com/package/npm) and [nodejs](https://nodejs.org/en/).
+
+#### 1. Clone the repository:
+git clone https://github.com/nolan-3/Dodgeball
```
In markdown, if you put your code inside triple backticks to format it like code:
````
```
git clone https://github.com/nolan-3/Dodgeball
```
````

Renders as:
```
git clone https://github.com/nolan-3/Dodgeball
```

Don't think too hard about how I wrote this comment (which is also in markdown).

nolan-3 committed:
saving a version from before I make Dodge into a class so I can go back if needed

nolan-3 committed:
Dodge is now a class component, but I'm not sure how
 to give the projectiles access to its information, I have tried using extend and bind (possibly
among other things which I forgot

nolan-3 committed:
Resolved most comments

nolan-3 committed:
fixed the logic for intersection between circle and rectangle, made 2 intersection functions and moved them into Intersections.js

nolan-3 committed:
I should have some time to look at models tomorrow

nolan-3 committed:
start of a battle screen

nolan-3 committed:
start of a battle screen

nolan-3 committed:
start of a battle screen

nolan-3 committed:
beginning of health bars

nolan-3 committed:
created a reasonably working healthbar, still to add animations, dodgeball passes a score to the Battle screen which calculates damage and passes it back

nolan-3 committed:
health bars now can change independently, deleted results.js

nolan-3 committed:
deleted references to results.js :)
